ITISFoundation/osparc-simcore
# Is1233/auto update postgres

## What do these changes do?


- [x] adds a postgres migration docker container for use in local machine when initializing the database, or upgrading the database
- [x] adds a Makefile recipe to upgrade the database from the command line ```make postgres-upgrade```

Behind the scene, the container has the **sc-pg** package from github (current branch, not LOCAL changes). then the service is attached to the simcore network, and by default runs ```sc-pg upgrade```

## Related issue number


#1233 
@ignapas please check what else is necessary. I think this closes the issue as far as I know.

## How to test


```
make leave # ensure nothing is running
docker volume prune # ensure the database volume is removed from host
make build # ensure the latest and greatest is available
make up-prod # runs the latest build
# now have a look in [http://127.0.0.1:18080](http://127.0.0.1:18080), the database is empty
make postgres-upgrade # will build the migration container and run it straight away
# now have a look again in [http://127.0.0.1:18080](http://127.0.0.1:18080), the database is initialized/upgraded
```
## Checklist
